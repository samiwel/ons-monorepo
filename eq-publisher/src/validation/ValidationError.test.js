const ValidationError = require("./ValidationError");

describe("validation error", () => {
  it("should accept message", () => {
    const validationError = new ValidationError("message", null);
    expect(validationError.message).toEqual("message");
  });

  it("should accept a validation object", () => {
    const mockValidationObject = {
      validationErrors: [
        {
          error: "some error",
          description: "some description"
        }
      ]
    };

    const validationError = new ValidationError(
      null,
      null,
      mockValidationObject
    );

    expect(validationError.result).toBe(mockValidationObject);
  });

  it("should override toString", () => {
    const mockValidationObject = {
      validationErrors: [
        {
          error: "some error",
          description: "some description"
        }
      ]
    };

    const validationError = new ValidationError(
      "Error message",
      {},
      mockValidationObject
    );
    const errorMessage = validationError.toString();

    expect(errorMessage).toContain("Error message");
    expect(errorMessage).toContain(JSON.stringify(mockValidationObject));
  });

  it("should override toJSON", () => {
    const mockValidationObject = {
      validationErrors: [
        {
          error: "some error",
          description: "some description"
        }
      ]
    };

    const mockQuestionnaire = {
      id: 1,
      title: "hello world"
    };

    const validationError = new ValidationError(
      "Something has gone terribly wrong",
      mockQuestionnaire,
      mockValidationObject
    );

    expect(JSON.stringify(validationError)).toMatchSnapshot();
  });
});
