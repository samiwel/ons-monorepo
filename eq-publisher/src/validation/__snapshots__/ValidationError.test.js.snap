// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation error should override toJSON 1`] = `"{\\"message\\":\\"Something has gone terribly wrong\\",\\"validatorInput\\":{\\"id\\":1,\\"title\\":\\"hello world\\"},\\"validatorResult\\":{\\"validationErrors\\":[{\\"error\\":\\"some error\\",\\"description\\":\\"some description\\"}]}}"`;
